zcash_client_backend/src/data_api.rs

```diff
@@ -1542,6 +1621,28 @@ pub trait WalletWrite: WalletRead {
     ///
     /// There may be restrictions on heights to which it is possible to truncate.
     fn truncate_to_height(&mut self, block_height: BlockHeight) -> Result<(), Self::Error>;
+
+    /// Reserves the next `n` available ephemeral addresses for the given account.
+    /// This cannot be undone, so as far as possible, errors associated with transaction
+    /// construction should have been reported before calling this method.
+    ///
+    /// To ensure that sufficient information is stored on-chain to allow recovering
+    /// funds sent back to any of the used addresses, a "gap limit" of 20 addresses
+    /// should be observed as described in [BIP 44].
+    ///
+    /// Returns an error if there is insufficient space within the gap limit to allocate
+    /// the given number of addresses, or if the account identifier does not correspond
+    /// to a known account.
+    ///
+    /// Precondition: `n >= 0`
```

Comment:
Why is 0 a valid value?

Reply:
There's no reason to exclude it; you're allocating 0 addresses which always succeeds.

Reply:
Does the calling code enforce that the number of addresses returned is equal to the number of addresses that was requested?

Reply:
Nonblocking, but I really don't see why we need to allocate more than one at a time. It adds error cases that need to be handled; if the method only returned a single address, then we could make failure to generate the address an error; it would be able to return `Result<(TransparentAddress, TransparentAddressMetadata), Self::Error>` directly.

Reply:
> Does the calling code enforce that the number of addresses returned is equal to the number of addresses that was requested?

Yes.

> Nonblocking, but I really don't see why we need to allocate more than one at a time. It adds error cases that need to be handled; if the method only returned a single address, then we could make failure to generate the address an error; it would be able to return `Result<(TransparentAddress, TransparentAddressMetadata), Self::Error>` directly.

You may be right, but it works as-is and I don't think it would be much of a simplification.